src/deepsparse/pipeline_loggers/multi_logger.py

```diff
@@ -0,0 +1,124 @@
+from deepsparse.pipeline_loggers.prometheus_pipeline_logger import PrometheusLogger
+from deepsparse.timing.timing_schema import InferenceTimingSchema
+
+
+__all__ = ["MultiLogger"]
+
+LOGGERS = [PrometheusLogger]
+SUPPORTED_LOGGERS = {logger.identifier: logger for logger in LOGGERS}
+
+
+class MultiLogger(PipelineLogger):
+    """
+    Object that contains multiple loggers for
+    the given inference pipeline.
+
+    The envisioned lifecycle of a logger
+    manager:
+
+    ```
+    pipeline = ... # define a pipeline
+    pipeline_name = pipeline.name # fetch the pipeline name
+
+    # create a MultiLogger
+    logger_manager = MultiLogger("logger_name")
+    or
+    logger_manager = MultiLogger(["logger_name_1", "logger_name_2", ...])
+
+    # log the data for the particular inference pipeline
+
+    timings, data = ...
+    logger.log_timings(pipeline_name, timings)
+    logger.log_data(pipeline_name, data)
+
+    ```
+
+    :param logger_identifiers: The identifier(s) of the logger types that
+        will be created within the scope of the manager. This can be either a
+        single identifier, or a list of identifiers (for multiple loggers).
+    """
+
+    def __init__(
+        self,
+        logger_identifiers: Union[str, List[str]],
+    ):
+        self._supported_loggers = SUPPORTED_LOGGERS
+        self._loggers = self._setup(logger_identifiers)
+        self._start()
+
+    @property
+    def loggers(self) -> Dict[str, PipelineLogger]:
+        """
+        :return: The mapping from the logger identifier
+        to the PipelineLogger instance
+        """
+        return self._loggers
+
+    def log_latency(self, pipeline_name: str, inference_timing: InferenceTimingSchema):
+        """
+        Logs the inference latency information to the all the loggers in manager's scope
+
+        :param pipeline_name: The name of the inference pipeline from which the
+            logger consumes the inference information to be monitored
+        :param inference_timing: Pydantic model that contains information
+            about time deltas of various processes within the inference pipeline
+        """
+        for name, logger in self.loggers.items():
+            logger.log_latency(pipeline_name, inference_timing)
+
+    def log_data(self, pipeline_name: str, inputs: Any, outputs: Any):
+        """
+        Logs the inference inputs and outputs to all the loggers in manager's scope
+
+        :param pipeline_name: The name of the inference pipeline from which the
+            logger consumes the data information to be monitored
+        :param inputs: the data received and consumed by the inference
+            pipeline
+        :param outputs: the data returned by the inference pipeline
+        """
+        for name, logger in self.loggers.items():
+            logger.log_data(pipeline_name, inputs, outputs)
+
+    def _setup(
+        self, logger_identifiers: Union[str, List[str]]
```

Comment:
need to think if we can do something more extensible here than string names to the desired loggers - ie if there's stuff we need to configure with an integration like prometheus's port, save directory, etc - we'd like to be able to control that at the server (and even user) level. 

so just taking a list of constructed logger objects here and ignoring identifiers might be the way to go, leaving the onus of construction to the server

Reply:
good point!